addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1563,25 +1563,28 @@ export class Wysiwyg extends Component {
         const targetEl = this.odooEditor.document.getSelection();
         const closest = closestBlock(targetEl.anchorNode);
         const restoreSelection = preserveCursor(this.odooEditor.document);
-        const blockWidth = parseInt(getComputedStyle(closest).width.slice(0, -2)); // remove 'px'
-        // Calculate one/fouth portion of blockWidth, and then on the basis of our cursor position
-        // check in which quadrant it falls, give the popover its position accordingly
-        const portionWidth = (blockWidth / 4);
-        const cursorAt = getRangePosition(targetEl, this.odooEditor.document).left;
-        let position = 'end';
-        if (cursorAt < portionWidth * 2) {
-            position = 'start';
-        } else if (cursorAt <= portionWidth * 3) {
-            position = 'middle';
-        }
```

Comment:
This seems like it would be useful. Why did you remove it?

Reply:
It's used to determine the popover's position based on the cursor position, but it has limitations, as it only supports three fixed positions (start, middle, end) of block. 
and that is why I have used `onPositioned` in which the popover positioned at cursor position.
So, we no longer require this as the popover position will be determined within the `onPositioned`.

Reply:
But didn't this ensure the popover couldn't get out of bounds of the viewport?

Reply:
Yes, this code ensures the popover stays within the `viewport`. However, even if I don't remove this code, the popover position set by it will be overwritten by my changes.

Additionally, I haven't encountered any situations where the popover goes out of bounds. I'm using the same positioning function as the `powerbox`, and since the `powerbox` doesn't have any issues like this, I believe the popover should be fine too.

Reply:
I tested and had no issues indeed. Thanks for the clarifications :+1: